woodwork/type_sys/type_system.py

```diff
@@ -315,11 +315,22 @@ def get_inference_matches(types_to_check, series, type_matches=[]):
                 get_inference_matches(check_next, series, type_matches)
             return type_matches
 
-        type_matches = get_inference_matches(self.root_types, series)
+        # Don't include NaturalLanguage as we only want to check that if
+        # no other matches are found
+        types_to_check = [
+            ltype for ltype in self.root_types if ltype != NaturalLanguage
+        ]
+        type_matches = get_inference_matches(types_to_check, series)
 
         if len(type_matches) == 0:
-            # If no matches, set type to default type (Unknown)
-            logical_type = self.default_type
+            # Check if this is NaturalLanguage, otherwise set
```

Comment:
should we leave a comment explaining that we do this because the inference function was created on these assumptions and for performance reasons?

Reply:
Updated the comment to better reflect this.